lms/static/scripts/frontend_apps/components/FilePickerApp.js

```diff
@@ -94,13 +98,39 @@ export default function FilePickerApp({ onSubmit }) {
   // Submit the form after a selection is made via one of the available
   // methods.
   useEffect(() => {
+    async function createAssignment() {
+      /** @type Record<string, string|Content|null> */
+      const data = { ...createAssignmentAPI.data };
+
+      data.content = content;
+      data.groupset = groupConfig.groupSet;
+      const assignment = await apiCall({
```

Comment:
This doesn't have an expected error response, eg a re-authorization.

Should I then try/catch it? If that's the case how can I show a generic, something went wrong message? (or something more usefull, but this case would likely be a bug).

Should I have boolean `creatingAssignment` or similar between this an getting the answer?

Reply:
> Should I then try/catch it? If that's the case how can I show a generic, something went wrong message? (or something more usefull, but this case would likely be a bug).

Yes. There is an existing `errorInfo` state in this component that shows a dialog when set. We can probably re-use that here.

> Should I have boolean creatingAssignment or similar between this an getting the answer?

Does this call involve any calls to potentially slow third-party APIs or does all the work happen in our app. If the former, we probably should add a loading state. If the latter, it may be less important.

Reply:
Added a try and used `errorInfo`. This is how the UI looks:

![Screenshot from 2021-09-23 15-07-43](https://user-images.githubusercontent.com/1433832/134512506-5cd90ccd-deb5-40c2-9603-96b5711e9c89.png)